docs/develop/understanding-the-javascript-api-for-office.md

```diff
@@ -64,15 +124,28 @@ Office.initialize = function (reason) {
     });
  };
 ```
-For more information, see [Office.initialize Event](https://dev.office.com/reference/add-ins/shared/office.initialize) and [InitializationReason Enumeration](https://dev.office.com/reference/add-ins/shared/initializationreason-enumeration). 
+
+For more information, see [Office.initialize Event](https://dev.office.com/reference/add-ins/shared/office.initialize) and [InitializationReason Enumeration](https://dev.office.com/reference/add-ins/shared/initializationreason-enumeration).
+
+### Major differences between Office.initialize and Office.onReady()
+
+- You can assign only one handler to `Office.initialize` and it is called, only once, by the Office infrastructure. But you can call `Office.onReady()` in different places in your code and use different callbacks. For example, your code could call `Office.onReady()` as soon as your custom script loads (in, for example, an Immediately Invoked Function Expression) with a callback that runs initialization logic. But you could also have a button in the task pane, whose script calls `Office.onReady()` with a different callback. If so, the second callback runs when the button is clicked.
+
+- The `Office.initialize` event fires at the end of the internal process in which Office.js initializes itself. And it fires immediately after the internal process ends. If the code in which you assign a handler to the event executes too long after the event fires, then your handler doesn't run. For example, if you are using the WebPack task manager, it might configure the add-in's home page to load polyfill files after it loads Office.js but before it loads your custom JavaScript. By the time your script loads and assigns the handler, the initialize event has already happened. But it is never "too late" to call `Office.onReady()`. If the initialize event has already happened, the callback runs immediately.
+
+> [!NOTE]
+> Even if you have no start-up logic, it is a good practice to either call `Office.onReady()` or assign an empty function to `Office.initialize` when your add-in JavaScript loads, because some Office host and platform combinations won't load the task pane until one of these happens. The following line is often used:
+>
+>```js
+>Office.initialize = function () {};
```

Comment:
Should we make the sample use onReady, since that's what we're officially recommending?